lib/mpool/src/mpool_file.c

```diff
@@ -154,7 +169,7 @@ mpool_file_read(struct mpool_file *file, off_t offset, char *buf, size_t buflen,
     iov.iov_base = buf;
     iov.iov_len = buflen;
 
-    err = file->io.read(file->fd, offset, (const struct iovec *)&iov, 1, 0, rdlen);
+    err = file->io.read(NULL, file->fd, offset, (const struct iovec *)&iov, 1, 0, rdlen);
```

Comment:
The memcpy path in io_pmem_read() can be gone.. I initially added it to be symmetric with io_pmem_write(), but then realized that there's no use-case in HSE for it. Forgot to remove it.

The pmem_memcpy path in io_pmem_write() is meant for future use-cases like memory-mapped mblock writes (for cN) and mpool file writes (for WAL). I changed WAL to use this path, but didn't see much benefits due to its large sequential writes. I visualize that it will be useful when we will support durable transactions..

Let me actually revert both, it can be added later when needed.